### PR TITLE
MINOR: Fix a few more compiler warnings

### PR DESCRIPTION
**Pull request by original-brownbear**
Just stuff the compiler warns us about should be self explainatory with the exception of this part:

```diff
-        } else if (list instanceof ConvertedList) {		
-            clone = new ArrayList<>();		
         } else {		
```

This is simply never hit because `ConvertedList` is a child of `ArrayList` which is caught by the `instanceof ArrayList` check above => removed the dead code :)

and the change to `public static Event[] fromJson(String json)`:

* We should really use hard types here instead of `instanceof` checks for 2 reasons:
   * more compile time safety
   * `ClasscastException` with a proper is a lot easier to interpret that our homebrew `IOException`

Fixed about 30 warnings for #7701 :)

### CODE REVIEW

File: logstash-core/src/main/java/org/logstash/instrument/witness/pipeline/PluginsWitness.java [hunk 1]
```diff
@@ -89,8 +89,8 @@ public void forgetAll() {
      * @param id     the id of the plugin
      * @return existing or new {@link PluginWitness}
      */
-    private PluginWitness getPlugin(Map<String, PluginWitness> plugin, String id) {
-        return plugin.computeIfAbsent(id, k -> new PluginWitness(k));
+    private static PluginWitness getPlugin(Map<String, PluginWitness> plugin, String id) {
+        return plugin.computeIfAbsent(id, PluginWitness::new);
```
Comment (jakelandis) [inline logstash-core/src/main/java/org/logstash/instrument/witness/pipeline/PluginsWitness.java]: Was this actually a compiler warning, or just an IntelliJ default inspection warning ?

File: logstash-core/src/main/java/org/logstash/Cloner.java [hunk 2]
```diff
@@ -1,6 +1,13 @@
...
 
-import java.util.*;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
```
Comment (jakelandis) [inline logstash-core/src/main/java/org/logstash/Cloner.java]: Is this a compiler warning, or a style change ?

File: logstash-core/src/main/java/org/logstash/Event.java [hunk 3]
```diff
@@ -56,7 +57,7 @@ public Event()
      * @param data Map that is assumed to have either {@link String} or {@link RubyString}
      * keys and may contain Java and Ruby objects.
      */
-    public Event(Map data) {
+    public Event(final Map<String, Object> data) {
```
Comment (jakelandis) [inline logstash-core/src/main/java/org/logstash/Event.java]: The java doc claims either String or RubyString.  Does this break that contract ?

File: logstash-core/src/main/java/org/logstash/Event.java [hunk 4]
```diff
@@ -209,15 +211,12 @@ public String toJson() throws JsonProcessingException {
...
         } else if (o instanceof List) {
-            result = new Event[((List) o).size()];
+            result = new Event[((Collection<?>) o).size()];
             int i = 0;
-            for (Object e : (List)o) {
-                if (!(e instanceof Map)) {
-                    throw new IOException("incompatible inner json array object type=" + e.getClass().getName() + " , only hash map is supported");
-                }
-                result[i++] = new Event((Map)e);
+            for (Map<String, Object> e : (Iterable<Map<String, Object>>) o) {
```
Comment (jakelandis) [inline logstash-core/src/main/java/org/logstash/Event.java]: While correct, the code is a bit hard to read. We check if it is List, then upcast to a Collection , the iterate over it as a Iterable.  Can we just cast to List<?> once and use that variable through out ?

File: logstash-core/src/main/java/org/logstash/Event.java [hunk 5]
```diff
@@ -226,13 +225,13 @@ public String toJson() throws JsonProcessingException {
...
-    public Map toMap() {
+    public Map<String, Object> toMap() {
         return Cloner.deep(this.data);
     }
 
     public Event overwrite(Event e) {
-        this.data = e.getData();
-        this.cancelled = e.isCancelled();
+        this.data = e.data;
+        this.cancelled = e.cancelled;
```
Comment (jakelandis) [inline logstash-core/src/main/java/org/logstash/Event.java]: curious to why this change?  Don't we want to encourage the standard getter pattern ? (is this a compiler warning ?)

File: logstash-core/src/main/java/org/logstash/Event.java (hunk 4 above)
Comment (original-brownbear) [reply logstash-core/src/main/java/org/logstash/Event.java]: @jakelandis yea def! Will fix this soon :)

File: logstash-core/src/main/java/org/logstash/Event.java (hunk 3 above)
Comment (original-brownbear) [reply logstash-core/src/main/java/org/logstash/Event.java]: @jakelandis urgh you're right ... we could still have `RubyString` here, but type erasure prevents this from throwing obviously :( 
=> gotta change this to `? extends Serializable`

File: logstash-core/src/main/java/org/logstash/Cloner.java (hunk 2 above)
Comment (original-brownbear) [reply logstash-core/src/main/java/org/logstash/Cloner.java]: style change :) `.*` imports are just annoying

File: logstash-core/src/main/java/org/logstash/instrument/witness/pipeline/PluginsWitness.java (hunk 1 above)
Comment (original-brownbear) [reply logstash-core/src/main/java/org/logstash/instrument/witness/pipeline/PluginsWitness.java]: @jakelandis hah seems like this was an idea bug, it didn't understand the type of `k` correctly in the EAP. Still much nicer this way imo :P

File: logstash-core/src/main/java/org/logstash/Event.java (hunk 5 above)
Comment (original-brownbear) [reply logstash-core/src/main/java/org/logstash/Event.java]: @jakelandis wops that randomly made it in here :) still, reduces the bytecode size of this method a little and since `Event` is `final` (-> getters provide no encapsulation in the context of `Event` itself) this is a fine cleanup imo :)

**Comment by original-brownbear** [comment]
@jakelandis all either answered or fixed :)

**Comment by jakelandis** [comment]
LGTM
